### DISCUSSION TITLE
fixing up INIT and loadup files

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#75] [configurations and releases]
this issue is an after-the-fact placeholder for issues I was working around INITfiles and loadup scripts.
not all resolved per @Anzus @nbriggs @rmkaplan (adopt INIT.MAIKO as a model):

 On Dec 6, 2020, at 5:31 PM, Ron Kaplan <ron.kaplan@post.harvard.edu> wrote:
>
> Ahh, I didn’t notice those library/INIT files during my cleanup, since they were consistent everywhere.
>
> I think those should be moved from library to greetfiles, in our new arrangement.  (But I’m not gitty enough to do it.)
>
> Note that all the stuff about the different subdirectories of fonts is obsolete, they are now pretty much lumped together.  I think the subdirectories had to do with the distribution of fonts onto floppies.

Yes, the version that we put out there should match where we expect the fonts to be installed.
>
> In the INIT.MAIKO, there are a few lines that don’t appear in later versions (including the one that I developed over the last few years).  These are the settings for IDLE.PROFILE (authentication and SAVEVM), the DST start/stop dates, the DEFAULTOSTYPE, and the short and long host names.
>
> Are they still useful/necessary?

I don't believe that "authentication" works, SAVEVM is still possibly OK.  The DST start/stop dates are wrong, as far as I know (we can find what they should be, though.)  The DEFAULTOSTYPE is good, and the short and long hostnames are things that a user might want to set up, so they should at least be mentioned though they're unlikely to be set to meaningful values in the distributed file.

>
>
>> On Dec 6, 2020, at 11:03 AM, Nick Briggs <nicholas.h.briggs@gmail.com> wrote:
>>
>> It's library/INIT.MAIKO in my collection of stuff
>>
>>
>>> On Dec 6, 2020, at 11:02 AM, Arun Welch <welch@anzus.com> wrote:
>>>
>>> <INIT.LISP>
>>
>

**Comment by masinter** [comment]
@rmkaplan 

> There is an environment variable LOGINDIR that tells medley where you want internal (CNDIR) or cd to go by default and where medley thinks you are initially connected to (which is distinct from your shell connected directory at the time your start up medley).  By that also says where your personal greeting file is (and your personal greeting file could then redefine that variable for later cd/CNDIR defaulting).  I presume that Larry’s runmedley defaults that variable if it isn’t already defined in the environment.
> So, that’s how it knows where your personal greeting file is, and what cd with no arguments goes to.  I think all the other file operations go to the (medley-internal, not the shell) connected directory (e.g. after cd foo or (CNDIR ‘foo), (SYSOUT ‘X.SYSOUT) will go to foo/X.sysout)).
> I have never used CHDIR, didn’t know it existed.  Seems to be changing the connected directory in a shell, but I’m not sure which shell it is or why one would want to do that.  Can someone explain?
> The location of the lisp.virtualmen is determined separately, defaults to my Mac home directory (/Users/kaplan).
@nbriggs 
> Underneath the Lisp stuff (which I haven't looked at in detail yet), there's the whole maze of twisty little passages in {dsk} vs {unix}.
> At the lisp level there's the lispusers/CD package, which is how Henry Thompson thought things should work between 1982-88, and VANILLA-INIT and GENERIC-INIT which also adjust things to how Stan Lanning thought it should work when he was mostly using it as a common lisp environment.
@nbriggs 
> The C code looks for $HOME/lisp.virtualmem on startup if you're running under X (or `~/lisp.virtualmem` if you're using a raw display device (!)) if there is: no explicit sysout name given on the command line, or LDESRCESYSOUT or LDESOURCESYSOUT environment variable in order, note spelling difference).

For saving an image, the C code says:
```
/************************************************************************/
/*                                                                      */
/*                      v m e m _ s a v e 0                             */
/*                                                                      */
/*      Implements the VMEMSAVE subr.  Write out the current lisp       */
/*      lisp image to the specified file.  If the sysout file-name      */
/*      is explicitly specified, the directory on which the file        */
/*      resides is exactly (?) an existing directory.  This is          */
/*      guaranteed by the Lisp code, \MAIKO.CHECKFREEPAGE in LLFAULT.   */
/*                                                                      */
/*      If the file argument is nill, the default sysout-file name,     */
/*      "~/lisp.virtualmem", is used, subject to override by the        */
/*      LDEDESTSYSOUT UNIX-environment variable.                        */
/*                                                                      */
```

I haven't tracked down where it decides that {DSK} is exactly your login home directory -- I do know that it's not based on $HOME because changing HOME doesn't affect it.

**Comment by masinter** [comment]
SAVEVM vs SYSOUT vs MAKESYS: these three Interlisp functions (should) call the same Maiko subr to save the VM.
They set up different variables and there are various BEFORExxxFORMS and AFTERxxxxFORMS.

Let's decide as a group what things SHOULD do in 2020 with GitHub and Docker and WSL etc.

**Comment by rmkaplan** [comment]
With respect to Henry’s lispusers/CD package, that has some bells and whistles, but I think a lot of the Unix stuff (../ ~/) is now built-in to CNDIR, which the cd command invokes.
